src/main/java/com/uber/cadence/converter/JsonDataConverter.java

```diff
@@ -128,6 +146,16 @@ public JsonDataConverter(Function<GsonBuilder, GsonBuilder> builderInterceptor)
     if (content == null) {
       return null;
     }
+    // Deserialize thrift values.
```

Comment:
Seems to be cleaner if we just have one try/catch block as above and put the switch logic inside.